KeyActions/src/main/java/org/vaadin/artur/client/KeyActionConnector.java

```diff
@@ -44,12 +42,6 @@ protected void triggerEvent(final KeyDownEvent event) {
 		Scheduler.get().scheduleDeferred(new ScheduledCommand() {
 			@Override
 			public void execute() {
-				if (getParent() instanceof BeforeShortcutActionListener) {
```

Comment:
I have a feeling that this needs to do `getParent().flush()` to be sure that the contents of e.g. a TextField is sent to the server before the key event